File: 167. Two Sum II - Input Array Is Sorted.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        int n=numbers.size();
        vector<int>res;
        int left=0,right=n-1;
        while(left<right){
            if(numbers[left]+numbers[right]==target){
                res={left+1,right+1};
                return res;
                
            }else if(numbers[left]+numbers[right]>target){
                right--;
            }else left++;
        }
        return res;
    }
               
};

File: 167. Two Sum II - Input Array Is Sorted.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        if (numbers == null || numbers.length<2) {
            return new int[] { 1, 2 };
        }
        int[] ans = new int[2];
        int left = 0, right = numbers.length - 1;

        while (left < right) {
            int mid = numbers[left] + numbers[right];

            if (mid == target) {
                ans[0] = left + 1;
                ans[1] = right + 1;
                break;
            } else if (target < mid) {
                right--;
            } else {
                left++;
            }
        }
        return ans;
    }
}
